datasets/plugins/batch/batch_dataset_plugin.py

```diff
@@ -54,53 +56,88 @@ def __init__(
+            self._path = str(value)
+        else:
+            self._path = value
 
-    def _get_path_filters_columns(
-        self, columns: Optional[ColumnNames] = None, run_id: Optional[str] = None
-    ) -> Tuple[str, Optional[list], Optional[Iterable[str]]]:
-        path = self._get_dataset_path()
+    def _get_filters_columns(
+        self,
+        columns: Optional[ColumnNames] = None,
+        run_id: Optional[str] = None,
+        partitions: Optional[dict] = None,
+    ) -> Tuple[Optional[list], Optional[Iterable[str]]]:
         read_columns = self._get_read_columns(columns)
-        filters = None
+        filters: Optional[List[Tuple]] = None
         query_run_id = run_id if run_id else self.run_id
         if query_run_id:
             filters = [("run_id", "=", query_run_id)]
-        return path, filters, read_columns
+
+        if partitions:
+            if filters is None:
+                filters = []
+            for key, value in partitions.items():
+                filters.append((key, "=", value))
+
+        return filters, read_columns
 
     def to_spark_pandas(
         self,
         columns: Optional[str] = None,
         run_id: Optional[str] = None,
         conf: Optional["SparkConf"] = None,
+        partitions: Optional[dict] = None,
         **kwargs,
     ) -> "ps.DataFrame":
-        sdf: SparkDataFrame = self.to_spark(columns=columns, run_id=run_id, conf=conf, **kwargs)
-        psdf: ps.DataFrame = sdf.to_pandas_on_spark(index_col=kwargs.get("index_col", None))
-        return psdf
+        return self.to_spark(
+            columns=columns, run_id=run_id, conf=conf, partitions=partitions, **kwargs
+        ).to_pandas_on_spark(index_col=kwargs.get("index_col", None))
 
     def to_pandas(
         self,
         columns: Optional[str] = None,
         run_id: Optional[str] = None,
         storage_format: str = "parquet",
+        partitions: Optional[dict] = None,
         **kwargs,
     ) -> pd.DataFrame:
         if not (self.mode & Mode.READ):
             raise InvalidOperationException(f"Cannot read because mode={self.mode}")
 
-        path, filters, read_columns = self._get_path_filters_columns(columns, run_id=run_id)
+        filters, read_columns = self._get_filters_columns(columns, run_id, partitions)
+        self.path = self._get_dataset_path()
+        _logger.info(f"to_pandas({self.path=}, {read_columns=}, {partitions=}, {run_id=}, {filters=})")
 
         df: pd.DataFrame
         if storage_format == "parquet":
-            df = pandas.read_parquet(
-                path,
+            df = pd.read_parquet(
+                self.path,
                 columns=read_columns,
                 filters=filters,
                 engine=kwargs.get("engine", "pyarrow"),
                 **kwargs,
             )
         elif storage_format == "csv":
-            df = pandas.read_csv(path, usecols=read_columns, **kwargs)
+            df = pd.read_csv(self.path, usecols=read_columns, **kwargs)
```

Comment:
how does filtering on partitions work for csv?

Reply:
`pandas.read_csv()` unfortunately doesn't support filtering on Hive compliant paths

This is one reason to contemplate removing `csv` support (until asked for).

Reply:
Is it doing it post read now in a where clause? 

Feels like it could make sense to remove it. I'd guess local is one of the main uses here?